### PR TITLE
create fat tests for OpenIdAuthenticationMechanismDefinition responseType

### PR DESCRIPTION
**Pull request by jimmy1wu**
add fat tests for OpenIdAuthenticationMechanismDefinition responseType

these tests try response type values for different flows (auth code, implicit, and hybrid) and makes sure that only the auth code flow passes as that is the only one the runtime currently supports. also tests for unknown response type value and empty value.

additionally, had to update the runtime, so an nls message is emitted if the runtime can't handle the response type.

example log:
```
[ERROR   ] CWWKS2423E: The following invalid response type is specified: 'id_token token'. Valid response types are ['code'].
```

### CODE REVIEW

File: dev/io.openliberty.security.oidcclientcore.internal/resources/io/openliberty/security/oidcclientcore/internal/resources/OidcClientCoreMessages.nlsprops [hunk 1]
```diff
@@ -111,3 +111,7 @@ JWK_CONNECTION_TIMED_OUT.useraction=Update the jwksConnectTimeout property in th
 JWK_READ_TIMED_OUT=CWWKS2422E: The OpenID Connect client failed to read data from the [{0}] JWK URI of the OpenID Connect provider within {1} milliseconds. Consider updating the jwksReadTimeout property in the OpenID Connect client configuration.
 JWK_READ_TIMED_OUT.explanation=The OpenID Connect client cannot read the JWK data from the JWK URI within the configured time frame.
 JWK_READ_TIMED_OUT.useraction=Update the jwksReadTimeout property in the OpenID Connect client configuration. Verify that the OpenID Connect provider is reachable.
+
+OIDC_CLIENT_INVALID_RESPONSE_TYPE=CWWKS2423E: Invalid response type ''{0}'' is specified. Valid response types are [{1}].
```
Comment (dmuelle) [inline dev/io.openliberty.security.oidcclientcore.internal/resources/io/openliberty/security/oidcclientcore/internal/resources/OidcClientCoreMessages.nlsprops]:
```suggestion
OIDC_CLIENT_INVALID_RESPONSE_TYPE=CWWKS2423E: The following invalid response type is specified: {0}. Valid response types are [{1}].
```

File: dev/io.openliberty.security.oidcclientcore.internal/resources/io/openliberty/security/oidcclientcore/internal/resources/OidcClientCoreMessages.nlsprops [hunk 2]
```diff
@@ -111,3 +111,7 @@ JWK_CONNECTION_TIMED_OUT.useraction=Update the jwksConnectTimeout property in th
...
+OIDC_CLIENT_INVALID_RESPONSE_TYPE.explanation=Specified response type is not valid.
```
Comment (dmuelle) [inline dev/io.openliberty.security.oidcclientcore.internal/resources/io/openliberty/security/oidcclientcore/internal/resources/OidcClientCoreMessages.nlsprops]:
```suggestion
OIDC_CLIENT_INVALID_RESPONSE_TYPE.explanation=The specified response type is not valid. 
```

File: dev/io.openliberty.security.oidcclientcore.internal/resources/io/openliberty/security/oidcclientcore/internal/resources/OidcClientCoreMessages.nlsprops [hunk 3]
```diff
@@ -111,3 +111,7 @@ JWK_CONNECTION_TIMED_OUT.useraction=Update the jwksConnectTimeout property in th
 JWK_READ_TIMED_OUT=CWWKS2422E: The OpenID Connect client failed to read data from the [{0}] JWK URI of the OpenID Connect provider within {1} milliseconds. Consider updating the jwksReadTimeout property in the OpenID Connect client configuration.
 JWK_READ_TIMED_OUT.explanation=The OpenID Connect client cannot read the JWK data from the JWK URI within the configured time frame.
 JWK_READ_TIMED_OUT.useraction=Update the jwksReadTimeout property in the OpenID Connect client configuration. Verify that the OpenID Connect provider is reachable.
+
+OIDC_CLIENT_INVALID_RESPONSE_TYPE=CWWKS2423E: Invalid response type ''{0}'' is specified. Valid response types are [{1}].
+OIDC_CLIENT_INVALID_RESPONSE_TYPE.explanation=Specified response type is not valid. 
+OIDC_CLIENT_INVALID_RESPONSE_TYPE.useraction=Specify one of the available response types.
```
Comment (dmuelle) [inline dev/io.openliberty.security.oidcclientcore.internal/resources/io/openliberty/security/oidcclientcore/internal/resources/OidcClientCoreMessages.nlsprops]:
```suggestion
OIDC_CLIENT_INVALID_RESPONSE_TYPE.useraction=Specify one of the valid response types.
```

**Review by dmuelle — CHANGES_REQUESTED** [review]
ID review

File: dev/io.openliberty.security.oidcclientcore.internal/resources/io/openliberty/security/oidcclientcore/internal/resources/OidcClientCoreMessages.nlsprops (hunk 1 above)
Comment (jimmy1wu) [reply dev/io.openliberty.security.oidcclientcore.internal/resources/io/openliberty/security/oidcclientcore/internal/resources/OidcClientCoreMessages.nlsprops]: hi @dmuelle , i am wondering if it'd be okay to do:
`The following invalid response type is specified: ''{0}''. Valid response types are [{1}].`

i wanted to keep the quotes (they are single quotes, but need to do two single quotes in a row for it to display as one) in there because the response type can have spaces and i am worried that not having the quote would be confusing.

i.e., 
**with quotes:**
```
The following invalid response type is specified: 'id_token token'. Valid response types are ['code','code token'].
```
vs.
**without quotes:**
```
The following invalid response type is specified: id_token token. Valid response types are [code,code token].
```

for the valid response types, i was planning on adding the quotes in the runtime, since it is a list.

note: the only valid one response type currently is `code`, but i added `code token` as well for the sake of the example.

Comment (dmuelle) [reply dev/io.openliberty.security.oidcclientcore.internal/resources/io/openliberty/security/oidcclientcore/internal/resources/OidcClientCoreMessages.nlsprops]: yes- if readability for the  response types is a concern, the quotes are ok. We generally try to avoid them but sometimes they are necessary

**Review by dmuelle — DISMISSED** [review]
LGTM, thanks

**Comment by jimmy1wu** [comment]
#libby-build
#spawn.fullfat.buckets=io.openliberty.security.jakartasec.3.0.internal_fat,io.openliberty.security.jakartasec.3.0.internal_fat.config

**Review by c00crane — DISMISSED** [review]

**Comment by jimmy1wu** [comment]
#build
#spawn.fullfat.buckets=io.openliberty.security.jakartasec.3.0.internal_fat,io.openliberty.security.jakartasec.3.0.internal_fat.config

**Comment by jimmy1wu** [comment]
#libby

**Review by c00crane — APPROVED** [review]
